pkg/asset-manager-utils/contracts/RewardsAssetManager.sol

```diff
@@ -206,6 +206,20 @@ abstract contract RewardsAssetManager is IAssetManager {
         return FixedPoint.mulDown(criticalManagedBalance.sub(poolManaged), config.feePercentage);
     }
 
+    /**
+     * @notice withdraw `amount` of cash from the Vault, reducing the pool's TVL
+     * @dev When withdrawing `amount` will be moved from the pool's cash to managed balance
+     * As these funds are to be paid as fees (and so lost) we cache the previous managed balance and restore it.
+     */
+    function _withdrawCashFromVault(uint256 amount) private {
+        // Pull funds from the vault and update balance to reflect that the fee is no longer part of managed funds
+        IVault.PoolBalanceOp[] memory ops = new IVault.PoolBalanceOp[](2);
+        ops[0] = IVault.PoolBalanceOp(IVault.PoolBalanceOpKind.WITHDRAW, poolId, token, amount);
+        ops[1] = IVault.PoolBalanceOp(IVault.PoolBalanceOpKind.UPDATE, poolId, token, readAUM().sub(amount));
```

Comment:
The comments and implementation don't match: we're not restoring the previous value, but rather syncing the Vault's `managed` value with the actual current value of AUM.